sdk/vppagent/memif_connect.go

```diff
@@ -0,0 +1,130 @@
+package vppagent
+
+import (
+	"context"
+	"os"
+	"path"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/configurator"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	"github.com/networkservicemesh/networkservicemesh/sdk/endpoint"
+	"github.com/sirupsen/logrus"
+)
+
+// MemifConnect is a VPP Agent Memif Connect composite
+type MemifConnect struct {
+	endpoint.BaseCompositeEndpoint
+	Workspace      string
+	ConnectionSide ConnectionSide
+	Connections    map[string]*ConnectionData
+}
+
+// Request implements the request handler
+func (mc *MemifConnect) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+
+	if mc.GetNext() == nil {
+		logrus.Fatal("The VPP Agent Memif Connect composite requires that there is Next set")
+	}
+
+	incomingConnection, err := mc.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	var connectionData *ConnectionData
+	opaque := mc.GetNext().GetOpaque(incomingConnection)
+	if opaque != nil {
+		connectionData = opaque.(*ConnectionData)
+		if connectionData.DataChange == nil {
+			connectionData.DataChange = &configurator.Config{}
+		}
+	} else {
+		connectionData = &ConnectionData{
+			DataChange: &configurator.Config{},
+		}
+	}
+
+	if connectionData.DataChange.VppConfig == nil {
+		connectionData.DataChange.VppConfig = &vpp.ConfigData{}
+	}
+
+	socketFilename := path.Join(mc.Workspace, incomingConnection.GetMechanism().GetSocketFilename())
+	socketDir := path.Dir(socketFilename)
+
+	if err := os.MkdirAll(socketDir, os.ModePerm); err != nil {
+		return nil, err
+	}
+
+	var name string
+	if mc.ConnectionSide == DESTINATION {
+		name = "DST-" + incomingConnection.GetId()
+	} else {
+		name = "SRC-" + incomingConnection.GetId()
```

Comment:
Src and Dst are just legacy from the old code. Let's replace these.

Reply:
Replace with in/out or eliminate at all?

Reply:
Removed